### ISSUE TITLE
windows/amd64: unable to debug minidumps created via GOTRACEBACK=wer

### ISSUE THREAD

**Issue by tuxxi** [go-delve/delve#3824]
Getting `error while reading spliced memory at ...: EOF` when attempting to debug a minidump created using the `GOTRACEBACK=wer` feature from Go 1.21+

1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.23.1
Build: $Id: 2eba762d75437d380e48fc42213853f13aa2904d $
```
2. What version of Go are you using? (`go version`)?
```
go version go1.23.2 windows/amd64
```
3. What operating system and processor architecture are you using?
```
Windows 11 (23H2 22631.4169) 
amd64
```
4. What did you do?
* Create a minidump by running a crashing application under `GOTRACEBACK=wer`
* Debug the minidump with `dlv core <exe> <dmp>` 
* Try to print stack trace, goroutines, etc
5. What did you expect to see?
* Expect to see stack trace, goroutines, etc. 
* `dlv core` works if I run the program under `dlv debug ...`, then within dlv run `dump myprog.dmp`, then finally use `dlv core`
6. What did you see instead?
```
(dlv) bt
0  0x00007ffde8af0612 in ???
   at ?:-1
   error: error while reading spliced memory at 0x9: EOF
(truncated)
(dlv) grs
Command failed: error while reading spliced memory at 0x603418: EOF
```

### Short reproducible example

(First, enable local dumps collection by creating reg key at `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\Windows Error Reporting\LocalDumps`
See https://learn.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps for more info)
Then:
```
> type main.go
package main

import "fmt"

func main() {
        var somethingnil *string
        fmt.Println("on no", *somethingnil)
}

> go build main.go
> set GOTRACEBACK=wer
> .\main.exe
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xb69614]
...
> dlv core .\main.exe %LOCALAPPDATA%\CrashDumps\main.exe.23592.dmp
Type 'help' for list of commands.
(dlv) bt
0  0x00007ffde8af0612 in ???
   at ?:-1
   error: error while reading spliced memory at 0x9: EOF
```

**Comment by aarzilli** [comment]
Can you upload somewhere both main.exe and the dmp file?

**Comment by tuxxi** [comment]
Sure, see attached example: 
[dlv-windows-amd64-minidump.zip](https://github.com/user-attachments/files/17284147/dlv-windows-amd64-minidump.zip)

Edited the original to add one more step - to enable local dumps collection you need to create the reg key `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\Windows Error Reporting\LocalDumps`

https://learn.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps

**Comment by aarzilli** [comment]
Did you set DumpType to 2 (full dump)?

**Comment by tuxxi** [comment]
Thanks for your quick reply. Hmm, I get some better behavior for `goroutines` and `vars` etc if DumpType is 2 (full dump)

However `stack` (bt) is still having trouble unwinding the call stack.
```
> dlv core --log --log-output minidump main.exe %LOCALAPPDATA%\main.exe.35928.dmp
...
2024-10-07T22:37:22-07:00 info kind=minidump,layer=core File flags: FileWithHandleData|FileIgnoreInaccessibleMemory|FileWithThreadInfo|FileWithFullMemoryInfo|FileWithFullMemory|FileWithUnloadedModules
...
(dlv) bt
0  0x00007ffde8af0612 in ???
   at ?:-1
   error: error while reading spliced memory at 0x9: EOF
(truncated)
```

It's not clear to me exactly what should work. `locals`, `args` also fail (`Command failed: unable to find function context`). Perhaps you could help me understand what is supposed to be supported in coredump mode on windows/amd64, and what is not?

Thank you again for your time :+1:

**Comment by aarzilli** [comment]
Presumably you simply aren't on a goroutine. Try selecting one with the
goroutine command.

**Comment by aarzilli** [comment]
I don't think there's anything to do for this.

**Comment by st-matskevich** [comment]
I have the same problem. I can confirm that the goroutine is selected, I guess dlv even selects one by default.
```
PS C:\Users\stmat\Documents\Projects\test\go-crash> dlv core .\go-crash.exe .\go-crash.exe.11408.dmp
Type 'help' for list of commands.
(dlv) bt
0  0x00007ffec0de3942 in ???
   at ?:-1
   error: error while reading spliced memory at 0x9: EOF
(truncated)
(dlv) goroutines
* Goroutine 1 - User: :0 ??? (0x7ffec0de3942) (thread 16888)
  Goroutine 2 - User: C:/Program Files/Go/src/runtime/proc.go:425 runtime.gopark (0xf43bee) [force gc (idle)]
  Goroutine 3 - User: C:/Program Files/Go/src/runtime/proc.go:425 runtime.gopark (0xf43bee) [GC sweep wait]
  Goroutine 4 - User: C:/Program Files/Go/src/runtime/proc.go:425 runtime.gopark (0xf43bee) [GC scavenge wait]
  Goroutine 5 - User: C:/Program Files/Go/src/runtime/proc.go:425 runtime.gopark (0xf43bee) [finalizer wait]
[5 goroutines]
(dlv) goroutine 1
Switched from 1 to 1 (thread 16888)
(dlv) bt
0  0x00007ffec0de3942 in ???
   at ?:-1
   error: error while reading spliced memory at 0x9: EOF
(truncated)
```

One thing I've noticed - if dump is generated manually via Task Manager, dlv can read it without any issues
```
PS C:\Users\stmat\Documents\Projects\test\go-crash> dlv core .\go-crash.exe .\go-crash.dmp
Type 'help' for list of commands.
(dlv) bt
0  0x0000000000459695 in main.main
   at C:/Users/stmat/Documents/Projects/test/go-crash/main.go:12
1  0x00000000003f6f7d in runtime.main
   at C:/Program Files/Go/src/runtime/proc.go:272
2  0x000000000042bcc1 in runtime.goexit
   at C:/Program Files/Go/src/runtime/asm_amd64.s:1700
```

The code I've used to reproduce this
```
package main

import (
	"runtime/debug"
)

func main() {
	debug.SetTraceback("wer")

	counter := 0
	for i := 0; i < 10000000000; i++ {
		counter += i
	}

	z := 0
	_ = 21 / z
}
```

WER settings
```
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\Windows Error Reporting\LocalDumps\go-crash.exe
DumpCount   REG_DWORD        3
DumpFolder  REG_EXPAND_SZ    C:\Users\stmat\Documents\Projects\test\go-crash
DumpType    REG_DWORD        2
```

go version - 1.23.0
dlv version - 1.23.0
dlv build -  $Id: e673f2da02185a1c2b6dea52842b0a8c7e7a03b9 

@aarzilli can we reopen the issue?

**Comment by aarzilli** [comment]
Can you post both the minidump file as well as the executable used to generate it?

**Comment by st-matskevich** [comment]
Sure, executable - [go-crash.zip](https://github.com/user-attachments/files/17842631/go-crash.zip)
Dump generated by panic - [go-crash.exe.17156.dmp](https://github.com/user-attachments/files/17842602/go-crash.exe.17156.dmp)
Dump generated by Task Manager - [go-crash.DMP](https://github.com/user-attachments/files/17842613/go-crash.DMP)

**Comment by aarzilli** [comment]
It's true that the problem is not with the goroutine selection here, unfortunately there is no easy solution. A kernel function has been called that doesn't preserve the BP chain, I don't think there's a way to unwind it short of understanding (and automatically downloading) windows PDB symbols.

**Comment by st-matskevich** [comment]
Ok, thank you for your investigation. Can we pin this information somehow? I wasted a few days trying to understand why crashdumps from our Windows service are useless.

Also, I don't quite understand how [WER CL](https://go.dev/cl/474915) ([go/#57441](https://github.com/golang/go/issues/57441)) got merged in the first place if produced crashdumps provide no information. Or does my case fall into some specific edge-case scenario?
